services/databaseEncryptionService.ts

```diff
@@ -23,55 +22,73 @@ export class EncryptionError extends Error {
 export const ERROR_CODES = {
   USER_CANCELLED: 'USER_CANCELLED',
   KEY_NOT_FOUND: 'KEY_NOT_FOUND',
+  UNINITIALIZED_ENCRYPTION: 'UNINITIALIZED_ENCRYPTION',
 } as const;
 
-let keyCache: Uint8Array | null = null;
+let keyCache: string | null = null;
 let initializationPromise: Promise<void> | null = null;
+let reWrapPromise: Promise<void> | null = null;
```

Comment:
Another promise? Is this really necessary?

EDIT: OK, I understand its purpose, which isn't what the name implies. I'd suggest `keyRewrappingPromise`.